Package.swift

```diff
@@ -1348,3 +1305,170 @@ func googleAppMeasurementDependency() -> Package.Dependency {
 
   return .package(url: appMeasurementURL, exact: "10.12.0")
 }
+
+func abseilDependency() -> Package.Dependency {
+  let packageInfo: (url: String, range: Range<Version>)
+
+  // If building Firestore from source, abseil will need to be built as source
+  // as the headers in the binary version of abseil are unusable.
+  if ProcessInfo.processInfo.environment["FIREBASE_SOURCE_FIRESTORE"] != nil {
+    packageInfo = (
+      "https://github.com/firebase/abseil-cpp-SwiftPM.git",
+      "0.20220623.0" ..< "0.20220624.0"
+    )
+  } else {
+    packageInfo = (
+      "https://github.com/google/abseil-cpp-binary.git",
+      "1.2022062300.0" ..< "1.2022062400.0"
+    )
```

Comment:
Previously, the binary abseil dep looked like:
```
    .package(
      url: "https://github.com/google/abseil-cpp-binary.git",
      "1.2022062300.0" ..< "1.2022062400.0"
    ),
```
So the URL and version range remains unchanged, as intended.